app/dashboard/utils.py

```diff
@@ -125,6 +135,30 @@ def create_user_action(user, action_type, request=None, metadata=None):
         return False
 
 
+def _get_utm_from_cookie(request):
+    """Extract utm* params from Cookie.
+
+    Args:
+        request (Request): The request object.
+
+    Returns:
+        utm_source: if it's not in cookie should be None.
+        utm_medium: if it's not in cookie should be None.
+        utm_campaign: if it's not in cookie should be None.
+
+    """
+    utm_source = request.COOKIES.get('utm_source')
+    utm_medium = request.COOKIES.get('utm_medium')
+    utm_campaign = request.COOKIES.get('utm_campaign')
+
+    try:
+        utm_campaign = json.loads(urllib.parse.unquote(utm_campaign))
+    except Exception:
+        utm_campaign = None
```

Comment:
Maybe we can plop all of these values into a dict here and return that versus all three unique vars? 🤔

Reply:
i just think they are same to use a dict or three vars. otherwise we want to store these values in one column in DB but it’s not good for analysis.